src/mock-doc/parser.ts

```diff
@@ -0,0 +1,16 @@
+import { MockDocument } from './document';
+import { parseHtmlToDocument } from './parse-html';
+
+export type DOMParserSupportedType =
+  | 'text/html'
+  | 'text/xml'
+  | 'application/xml'
+  | 'application/xhtml+xml'
+  | 'image/svg+xml';
+
+export class MockDOMParser {
+  parseFromString(string: string, type: DOMParserSupportedType): MockDocument {
```

Comment:
Since 'string' and 'type' are both technically reserved words in TypeScript, can we change these to use names that don't overlap with reserved keywords and are a little more descriptive?


```suggestion
  parseFromString(htmlToParse: string, mimeType: DOMParserSupportedType): MockDocument {
```